app/main.py

```diff
@@ -1,8 +1,62 @@
+from typing import List
+
+
 class Car:
-    # write your code here
-    pass
+    def __init__(
+            self,
+            comfort_class: int,
+            clean_mark: int,
+            brand: str
+    ) -> None:
+        self.comfort_class = comfort_class
+        self.clean_mark = clean_mark
+        self.brand = brand
 
 
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(
+            self,
+            distance_from_city_center: float,
+            clean_power: int,
+            average_rating: float,
+            count_of_ratings: int
+    ) -> None:
+        if distance_from_city_center == 0:
+            raise ValueError("Distance from city center cannot be zero")
+
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = average_rating
+        self.count_of_ratings = count_of_ratings
+
+    def serve_cars(self, cars: List[Car]) -> float:
+        income = 0.0
+        for car in cars:
+            if car.clean_mark < self.clean_power:
+                income += self.calculate_washing_price(car)
+                self.wash_single_car(car)
+
+        return income
+
+    def calculate_washing_price(self, car: Car) -> float:
+        clean_difference = self.clean_power - car.clean_mark
+        base_price = self.calculate_base_price(car)
+
+        return round(clean_difference * base_price, 1)
+
+    def calculate_base_price(self, car: Car) -> float:
+        return (
+            car.comfort_class * self.average_rating
+            / self.distance_from_city_center
+        )
+
+    def wash_single_car(self, car: Car) -> None:
+        car.clean_mark = self.clean_power
```

Comment:
Let's add `if` to check `clean_mark` and `clean_power` before washing car. What if we want to reuse this method in the future? We will add new `if` check again?

> wash_single_car - method, that washes a single car, so it should have clean_mark equals wash station's clean_power, if wash_station.clean_power is greater than car.clean_mark;